src/main/java/picard/analysis/CollectQualityYieldMetrics.java

```diff
@@ -77,74 +79,112 @@
+                                            final boolean includeSupplementalAlignments) {
+            this.useOriginalQualities          = useOriginalQualities;
+            this.includeSecondaryAlignments    = includeSecondaryAlignments;
+            this.includeSupplementalAlignments = includeSupplementalAlignments;
         }
 
-        // add up quals, and quals >= 20
-        for (final int qual : quals) {
-            metrics.Q20_EQUIVALENT_YIELD += qual;
-            if (qual >= 20) metrics.Q20_BASES++;
-            if (qual >= 30) metrics.Q30_BASES++;
+        public void acceptRecord(final SAMRecord rec, final ReferenceSequence ref) {
+            if (!this.includeSecondaryAlignments    && rec.getNotPrimaryAlignmentFlag()) return;
+            if (!this.includeSupplementalAlignments && rec.getSupplementaryAlignmentFlag()) return;
 
+            final int length = rec.getReadLength();
+            metrics.TOTAL_READS++;
+            metrics.TOTAL_BASES += length;
+
+            final boolean isPfRead = !rec.getReadFailsVendorQualityCheckFlag();
             if (isPfRead) {
-                metrics.PF_Q20_EQUIVALENT_YIELD += qual;
-                if (qual >= 20) metrics.PF_Q20_BASES++;
-                if (qual >= 30) metrics.PF_Q30_BASES++;
+                metrics.PF_READS++;
+                metrics.PF_BASES += length;
+            }
+
+            final byte[] quals;
+            if (this.useOriginalQualities) {
+                byte[] tmp = rec.getOriginalBaseQualities();
+                if (tmp == null) tmp = rec.getBaseQualities();
+                quals = tmp;
+            } else {
+                quals = rec.getBaseQualities();
+            }
+
+            // add up quals, and quals >= 20
+            for (final int qual : quals) {
+                metrics.Q20_EQUIVALENT_YIELD += qual;
+                if (qual >= 20) metrics.Q20_BASES++;
+                if (qual >= 30) metrics.Q30_BASES++;
+
+                if (isPfRead) {
+                    metrics.PF_Q20_EQUIVALENT_YIELD += qual;
+                    if (qual >= 20) metrics.PF_Q20_BASES++;
+                    if (qual >= 30) metrics.PF_Q30_BASES++;
+                }
             }
         }
-    }
 
-    @Override
-    protected void finish() {
-        final MetricsFile<QualityYieldMetrics, Integer> metricsFile = getMetricsFile();
-        metrics.READ_LENGTH = metrics.TOTAL_READS == 0 ? 0 : (int) (metrics.TOTAL_BASES / metrics.TOTAL_READS);
-        metrics.Q20_EQUIVALENT_YIELD = metrics.Q20_EQUIVALENT_YIELD / 20;
-        metrics.PF_Q20_EQUIVALENT_YIELD = metrics.PF_Q20_EQUIVALENT_YIELD / 20;
+        public void finish() {
+            metrics.READ_LENGTH             = metrics.TOTAL_READS == 0 ? 0 : (int) (metrics.TOTAL_BASES / metrics.TOTAL_READS);
+            metrics.Q20_EQUIVALENT_YIELD    = metrics.Q20_EQUIVALENT_YIELD / 20;
+            metrics.PF_Q20_EQUIVALENT_YIELD = metrics.PF_Q20_EQUIVALENT_YIELD / 20;
```

Comment:
I know this is not your doing...but could you correct the javadoc on "PF_Q20_EQUIVALENT_YIELD" (it's missing any mention of PF).